allennlp/training/metric_tracker.py

```diff
@@ -59,32 +59,31 @@ def clear(self) -> None:
         self._is_best_so_far = True
         self._epoch_number = 0
         self.best_epoch = None
+        self.best_epoch_metrics.clear()
 
     def state_dict(self) -> Dict[str, Any]:
         """
         A `Trainer` can use this to serialize the state of the metric tracker.
         """
         return {
             "best_so_far": self._best_so_far,
-            "patience": self._patience,
             "epochs_with_no_improvement": self._epochs_with_no_improvement,
             "is_best_so_far": self._is_best_so_far,
-            "best_epoch_metrics": self.best_epoch_metrics,
             "epoch_number": self._epoch_number,
             "best_epoch": self.best_epoch,
+            "best_epoch_metrics": self.best_epoch_metrics,
         }
 
     def load_state_dict(self, state_dict: Dict[str, Any]) -> None:
         """
         A `Trainer` can use this to hydrate a metric tracker from a serialized state.
         """
         self._best_so_far = state_dict["best_so_far"]
-        self._patience = state_dict["patience"]
         self._epochs_with_no_improvement = state_dict["epochs_with_no_improvement"]
         self._is_best_so_far = state_dict["is_best_so_far"]
-        self.best_epoch_metrics = state_dict["best_epoch_metrics"]
         self._epoch_number = state_dict["epoch_number"]
         self.best_epoch = state_dict["best_epoch"]
+        self.best_epoch_metrics = state_dict["best_epoch_metrics"]
```

Comment:
What if we try to `--recover` a run originally trained with previous versions? Do we need some kind of check here?

Reply:
That's not a supported scenario, though in this particular case it would be totally harmless to add a check.

Reply:
This is something that will be difficult to fix for a user. Even if it will only appear in v2.0, we may want backwards-compatibility here.

Reply:
Added in 764ce61.

Reply:
One more question: Is metric_tracker loaded when we run `allennlp evaluate`?

Reply:
No, only the trainer uses it.